lumigator/python/mzai/backend/backend/services/experiments.py

```diff
@@ -157,14 +157,30 @@ def get_experiment(self, experiment_id: UUID) -> ExperimentResponse:
         loguru.logger.info(f"Obtaining info for experiment {experiment_id}: {record}")
 
         all_succeeded = True
+        any_running = False
+        any_failed = False
+        # Any running: running
+        # Any failed: failed
+        # All succeeded: succeeded
         for job in self._get_all_owned_jobs(experiment_id):
             loguru.logger.info(f"Checking sub job: {job}")
             if self._job_service.get_job(job.id).status != JobStatus.SUCCEEDED:
                 all_succeeded = False
-                break
+                if self._job_service.get_job(job.id).status == JobStatus.FAILED:
+                    any_failed = True
+                    # Any failed job makes further inspection unnecessary
+                    break
+                if self._job_service.get_job(job.id).status == JobStatus.RUNNING:
+                    any_running = True
+                    # Any running job will move status to running, but
+                    # searching for failures is still necessary
 
         if all_succeeded:
             record = self._experiment_repo.update(experiment_id, status=JobStatus.SUCCEEDED)
+        if any_failed:
+            record = self._experiment_repo.update(experiment_id, status=JobStatus.FAILED)
+        if any_running:
+            record = self._experiment_repo.update(experiment_id, status=JobStatus.RUNNING)
```

Comment:
```suggestion
        elif any_failed:
            record = self._experiment_repo.update(experiment_id, status=JobStatus.FAILED)
        elif any_running:
            record = self._experiment_repo.update(experiment_id, status=JobStatus.RUNNING)
```

The first `elif` is probably not required (as the other variables are set only if at least one job is not SUCCEEDED`) but the latter is needed if we want to enforce the behavior you are setting above (see previous comment)

Reply:
There is a slight difference in semantics between `created` and `pending` (created happens at DB insertion time, pending after ray submission), even if I cannot think about a situation where ray accepted the job at submission time and then never even put it into pending state. I think the above should cover all cases (ie. we'll have `created` only until ray answers us with information about the job)

Reply:
I'm fine with the changes. I'll just re-run the tests with them and add documentation about the experiment states with these semantics 👍